packages/core/mesh/network-manager/src/swarm/peer.ts

```diff
@@ -250,27 +252,30 @@ export class Peer {
             initiator,
           });
 
-          if (this._lastConnectionTime && this._lastConnectionTime + CONNECTION_COUNTS_STABLE_AFTER < Date.now()) {
-            // If we're closing the connection, and it has been connected for a while, reset the backoff.
-            this._availableAfter = 0;
-          } else {
-            this.availableToConnect = false;
-            this._availableAfter = increaseInterval(this._availableAfter);
+          if (this.connection === connection) {
```

Comment:
This looks more as an invariant rather then a condition: given the comment below is there a situation where a new connection could be openned before this callback finishes processing?